worlds/blasphemous/__init__.py

```diff
@@ -273,133 +219,152 @@ def place_items_from_dict(self, option_dict: Dict[str, str]):
 
 
     def create_regions(self) -> None:
```

Comment:
You can optimize your create_regions function by not explicitly handling connections of exits and entrances and using the built in AP methods on `Region`. Here's an example that gives the same output as your code with same seed.

You can also avoid calling `location_table.index` by looping over an `enumerate` to get the index while looping, I added an example to the below code.

```py
    def create_regions(self) -> None:
        menu_region = Region("Menu", self.player, self.multiworld)
        misc_region = Region("Misc", self.player, self.multiworld)
        self.multiworld.regions += [menu_region, misc_region]

        for room in room_table:
            region = Region(room, self.player, self.multiworld)
            self.multiworld.regions.append(region)

        menu_region.add_exits({self.start_room: "New Game"})
        self.multiworld.get_region(self.start_room, self.player).add_exits({"Misc": "Misc"})

        for door in door_table:
            if door.get("OriginalDoor") is None:
                continue

            if not door["Id"] in self.door_connections.keys():
                self.door_connections[door["Id"]] = door["OriginalDoor"]
                self.door_connections[door["OriginalDoor"]] = door["Id"]

            parent_region = self.get_room_from_door(door["Id"])
            target_region = self.get_room_from_door(door["OriginalDoor"])
            parent_region.add_exits({
                target_region.name: door["Id"]
            }, {
                target_region.name: lambda x: door.get("VisibilityFlags") != 1
            })

        for loc_index, loc in enumerate(location_table):
            if not self.multiworld.boots_of_pleading[self.player] and loc["name"] == "BotSS: 2nd meeting with Redento":
                continue
            if not self.multiworld.purified_hand[self.player] and loc["name"] == "MoM: Western room ledge":
                continue

            region = self.multiworld.get_region(loc["room"], self.player)
            region.add_locations({loc["name"]: base_id + loc_index})

        for event_name, region_name in event_table.items():
            region = self.multiworld.get_region(region_name, self.player)
            event = BlasphemousLocation(self.player, event_name, None, region)
            event.show_in_spoiler = False
            event.place_locked_item(self.create_event(event_name))
            region.locations.append(event)

        for door in door_table:
            region = self.get_room_from_door(self.door_connections[door["Id"]])
            event = BlasphemousLocation(self.player, door["Id"], None, region)
            event.show_in_spoiler = False
            event.place_locked_item(self.create_event(door["Id"]))
            add_rule(event, lambda state: state.can_reach(self.get_connected_door(door["Id"])), self.player)
            region.locations.append(event)
        
        victory = Location(
            self.player, "His Holiness Escribar", None, self.multiworld.get_region("D07Z01S03", self.player))
        victory.place_locked_item(self.create_event("Victory"))
        self.multiworld.get_region("D07Z01S03", self.player).locations.append(victory)

        if self.multiworld.ending[self.player].value == 1:
            set_rule(victory, lambda state: state.has("Thorn Upgrade", self.player, 8))
        elif self.multiworld.ending[self.player].value == 2:
            set_rule(victory, lambda state: state.has("Thorn Upgrade", self.player, 8) and \
                state.has("Holy Wound of Abnegation", self.player))

        self.multiworld.completion_condition[self.player] = lambda state: state.has("Victory", self.player)
```